src/renderer/grid_renderer.rs

```diff
@@ -166,8 +166,9 @@ impl GridRenderer {
         if let Some(underline_style) = style.underline {
             let stroke_size = self.shaper.stroke_size();
             let underline_position = self.shaper.underline_position();
-            let p1 = pos + PixelVec::new(0.0, underline_position);
-            let p2 = pos + PixelVec::new(width, underline_position);
+            let underline_offset = SETTINGS.get::<RendererSettings>().underline_offset;
```

Comment:
I'm inclined to say this should be inside [the implementation of `underline_position`](https://github.com/neovide/neovide/blob/1db63d93b18cc1eb5ac5b3e67b3ca63505f10800/src/renderer/fonts/caching_shaper.rs#L213), and probably expressed as a fraction of `current_size`, so it scales with font size/screen DPI.

Reply:
I agree with both of these suggestions.

Reply:
Additionally, I think that it makes sense to ignore the font `underline_offset` offset completely if this is set, so that it completely overrides it.